### DISCUSSION TITLE
Exporting collector's own logs without going through a file

### DISCUSSION THREAD

**Discussion by chennin** [open-telemetry/opentelemetry-collector-contrib#17841] [Q&A]
I would like to send the collector's own logs to my  logging platform, but there does not seem to be a safe way to do this.

It is possible to send the logs to a file, then read that file and send it along as you would any other file. However, this seems inefficient (why can't I do this in memory?), and it leads to an infinite loop if the logging exporter's verbosity is set to `info` (!) or `debug` (in this case I am still on 0.55.0, but the problem would exist on newer versions with at least `detailed`). The collector emits a log line, reads it, then emits at least one log line, repeat forever. 

Is there a way to send the collector's own logs to another exporter without going through a file on disk?

**Comment by jpkrohling** [comment]
There was a similar issue for traces (and perhaps one for metrics?), and we will be addressing this soon. That said, I still believe that the collector should NOT be responsible for collecting its own telemetry ("who monitors the monitor" problem).
